### PR TITLE
B-21490/B-21492 - Secondary GBLOC

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Allows an admin use to assign up to two transportation offices to an office user.

Allows the office user to switch between the GBLOCs of those transportation offices.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

#### B-21492
1. Access the admin app.
2. Login as a new or existing admin user.
3. Open the form to create a new office user.
4. Fill out the form. Is the validation correct? Is the UI easy to use/clear?
5. Save a valid office user.
6. Repeat step 4 with the edit form.
7. Try to break it.

#### B-21490
1. Access the admin app
2. Create several office users of various roles each with two transportation offices. At least one should be a Services Counseling **Supervisor**.
3. Log in as each of the users, toggle between your GBLOCs with the dropdown in the top right, switch between queues, export your queue, just try to break it.
4. As your SC Supervisor verify the origin location dropdown filter's options update with the selected GBLOC.
5. **Optionally make one of your GBLOCs USMC as that is the branch most interested in this feature and their queues work a little differently (these changes don't modify how they work but it's likely worth testing the combination).**

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: pkg/services/transportation_office_assignments/transportation_office_assignments_updater.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+type transportaionOfficeAssignmentUpdater struct {
+}
+
+// NewTransportaionOfficeAssignmentUpdater creates a new struct with the service dependencies
+func NewTransportaionOfficeAssignmentUpdater() services.TransportaionOfficeAssignmentUpdater {
+	return transportaionOfficeAssignmentUpdater{}
+}
+
+func (updater transportaionOfficeAssignmentUpdater) UpdateTransportaionOfficeAssignments(
```
Comment (MInthavongsay) [inline pkg/services/transportation_office_assignments/transportation_office_assignments_updater.go:19]: Add unit test(s) for new service interfaces to increase test coverage.

**Review by MInthavongsay — APPROVED** [review]
Looks good. See comments about unit tests.

**Comment by antgmann** [comment]
Accepted Happo diffs

File: pkg/services/transportation_office_assignments/transportation_office_assignments_updater.go (hunk 1 above)
Comment (ajlusk) [reply pkg/services/transportation_office_assignments/transportation_office_assignments_updater.go:19]: Added

File: pkg/services/transportation_office_assignments/transportation_office_assignments_fetcher.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+type transportaionOfficeAssignmentFetcher struct {
+}
+
+// NewTransportaionOfficeAssignmentUpdater creates a new struct with the service dependencies
+func NewTransportaionOfficeAssignmentFetcher() services.TransportaionOfficeAssignmentFetcher {
+	return transportaionOfficeAssignmentFetcher{}
+}
+
+func (fetcher transportaionOfficeAssignmentFetcher) FetchTransportaionOfficeAssignmentsByOfficeUserID(
```
[the remark this answers is not in the document]
Comment (ajlusk) [reply pkg/services/transportation_office_assignments/transportation_office_assignments_fetcher.go:19]: Added

**Review by r-mettler — APPROVED** [review]

**Pull request merged**